### PR TITLE
Bug: Fixed binary string in the url of email verification

### LINKED ISSUE #1061: Fix binary string in the URL on the email verification
## Description
As a developer,
I need to verify my email on the VMS web application.
Need to fix the test cases related to this.

## Mocks
Locally, If you register to the VMS then from the command line you get the email verification link. If you put the email verification link to the web application then you will get an error due to a binary string in the URL. You have to fix that error.

[error link](https://user-images.githubusercontent.com/26095715/82704886-96353500-9c44-11ea-89e7-7e3a3f08daae.png)

## Acceptance Criteria
- We can verify the link without getting any error after this bug.
- Test cases related to this need to be fixed.

## Definition of Done
- [x] Approval by 1 mentor.

## Estimation
8-10 hours

### PR DESCRIPTION
**Pull request by decon-harsh**
The url's uid was a b string. Decoded the uid to string.  Removed some flake8 errors.

Fixes #1061 

# Type of Change:

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

# Mocks
Screenshot from 2020-09-04 00-10-37

Screenshot from 2020-09-04 00-10-46

# How Has This Been Tested?
Manually

# Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [X] My changes generate no new warnings

### CODE REVIEW

File: vms/registration/views.py [hunk 1]
```diff
@@ -90,7 +90,8 @@ def post(self, request):
                         })
                 try:
                     admin_country_name = request.POST.get('country')
-                    admin_country = Country.objects.get(name=admin_country_name)
+                    admin_country = Country.objects.get(
+                        name=admin_country_name)
```
Comment (satya7289) [inline vms/registration/views.py:94]: Here, nothing is changed.!

File: vms/registration/views.py [hunk 2]
```diff
@@ -283,7 +285,8 @@ def post(self, request):
                 if organization:
                     volunteer.organization = organization
                 else:
-                    unlisted_org = request.POST.get('vol-unlisted_organization')
+                    unlisted_org = request.POST.get(
+                        'vol-unlisted_organization')
```
Comment (satya7289) [inline vms/registration/views.py:289]: Here, also nothing is changed..!

**Review by satya7289 — CHANGES_REQUESTED** [review]
Please modified the test cases related to this.

File: vms/registration/views.py (hunk 1 above)
Comment (decon-harsh) [reply vms/registration/views.py:94]: Flake8 error was getting displayed ,words limit per line.

**Comment by decon-harsh** [comment]
@satya7289 Can you help me the test cases? 
Screenshot from 2020-09-05 12-49-31

This is the only test case I guess and it looks fine , what do I need to modify then?

**Review by satya7289 — CHANGES_REQUESTED** [review]
@decon-harsh here is the test case link, you have to uncomment these and run test case locally and fix if any error persits.
- [First test case](https://github.com/anitab-org/vms/blob/b58bc989bbe05c72dc431b7da99eff44a50548c1/vms/registration/tests/test_functional_admin.py#L649)
- [Second test case](https://github.com/anitab-org/vms/blob/b58bc989bbe05c72dc431b7da99eff44a50548c1/vms/registration/tests/test_functional_volunteer.py#L149)

**Comment by decon-harsh** [comment]
On it

**Comment by decon-harsh** [comment]
Done 

Screenshot from 2020-09-06 21-38-31

No errors

File: vms/registration/tests/test_functional_admin.py [hunk 3]
```diff
@@ -647,25 +647,25 @@ def test_ajax_check_states(self):
         self.assertEqual(response.status_code, 302)
 
     # TODO Fix the test case(regarding binary string in url)
```
Comment (satya7289) [inline vms/registration/tests/test_functional_admin.py:649]: Remove this line.

File: vms/registration/tests/test_functional_volunteer.py [hunk 4]
```diff
@@ -147,25 +147,25 @@ def test_null_values(self):
         self.assertEqual(len(blocks), 8)
 
     # TODO Fix the test case(regarding binary string in url)
```
Comment (satya7289) [inline vms/registration/tests/test_functional_volunteer.py:149]: Remove this line.

**Review by satya7289 — CHANGES_REQUESTED** [review]

File: vms/registration/tests/test_functional_volunteer.py (hunk 4 above)
Comment (decon-harsh) [reply vms/registration/tests/test_functional_volunteer.py:149]: Line 147?

Comment (satya7289) [reply vms/registration/tests/test_functional_volunteer.py:149]: Line 149, TODO part..!

**Comment by decon-harsh** [comment]
Okay this is weird , I almost made no change and the build failed !

**Comment by atharwa-24** [comment]
> 
> 
> Okay this is weird , I almost made no change and the build failed 

Same problem faced The build is getting failed. 
on #1078

**Comment by decon-harsh** [comment]
Lmao I re runned it and it passed, haha!

**Review by satya7289 — APPROVED** [review]
Looks good now.

**Review by SanketDG — DISMISSED** [review]
LGTM

**Review by isabelcosta — DISMISSED** [review]

**Comment by isabelcosta** [comment]
@anitab-org/qa-team can anyone test this?
@decon-harsh can you let us know how you tested this?

**Comment by decon-harsh** [comment]
@isabelcosta I ran all the test cases locally and checked it manually whether the correct link was getting generated . Images in mocks

**Comment by decon-harsh** [comment]
Hey @satya7289 just a question . Since I worked on the test cases in this issue , Doesn't that makes it a quality assurance too?Please correct me if I am wrong!

**Comment by satya7289** [comment]
> Hey @satya7289 just a question . Since I worked on the test cases in this issue , Doesn't that makes it a quality assurance too?Please correct me if I am wrong!

Agree.

**Comment by decon-harsh** [comment]
> > Hey @satya7289 just a question . Since I worked on the test cases in this issue , Doesn't that makes it a quality assurance too?Please correct me if I am wrong!
> 
> Agree.
Can you add the tag for the OSH plrease :)

**Comment by satya7289** [comment]
> > > Hey @satya7289 just a question . Since I worked on the test cases in this issue , Doesn't that makes it a quality assurance too?Please correct me if I am wrong!
> > 
> > 
> > Agree.
> 
> Can you add the tag for the OSH please :)

This issue is already a part of OSH.

**Comment by decon-harsh** [comment]
> > > > Hey @satya7289 just a question . Since I worked on the test cases in this issue , Doesn't that makes it a quality assurance too?Please correct me if I am wrong!
> > > 
> > > 
> > > Agree.
> > 
> > 
> > Can you add the tag for the OSH please :)
> 
> This issue is already a part of OSH.

I was talking about quality assurance tag

File: vms/registration/views.py (hunk 1 above)
Comment (devkapilbansal) [reply vms/registration/views.py:94]: @decon-harsh I don't think that the line is actually too long, you should revert it. flake8 shows this for line length greater than 88, but you should do it if there are more that one comma seaprated values here

File: vms/registration/views.py (hunk 2 above)
Comment (devkapilbansal) [reply vms/registration/views.py:289]: @decon-harsh revert this too

**Review by devkapilbansal — CHANGES_REQUESTED** [review]

**Comment by decon-harsh** [comment]
@devkapilbansal done please check

**Comment by devkapilbansal** [comment]
@decon-harsh codecov is not able to calculate test coverage as your branch is behind. Please rebase it to current develop. After that, I will test if it works properly and it can be labelled as Ready to Merge then

**Comment by decon-harsh** [comment]
Messed up? @devkapilbansal  

What should I do now?

**Comment by devkapilbansal** [comment]
@decon-harsh you should move back to the commit you do 3 hours before
```
git reset --hard c237667
git push -f
```

**Comment by decon-harsh** [comment]
other errors are coming up , let me close this pull and open another!
@devkapilbansal

**Pull request closed without merging**